src/nbl/video/COpenGLCommandBuffer.h

```diff
@@ -542,14 +542,40 @@ class COpenGLCommandBuffer final : public IGPUCommandBuffer
 
     COpenGLCommandBuffer(core::smart_refctd_ptr<const ILogicalDevice>&& dev, E_LEVEL lvl, core::smart_refctd_ptr<IGPUCommandPool>&& _cmdpool, system::logger_opt_smart_ptr&& logger, const COpenGLFeatureMap* _features);
 
+    void releaseResourcesBackToPool()
+    {
+        freeSpaceInCmdPool();
+        m_commands.clear();
+    }
+
+    bool resetCommon() override
+    {
+        // TODO resources for all command buffers already got freed from the command pool,
+        // this function should just make sure the command buffer stops pointing to memory that has
+        // been freed from the pool
+        // (this currently causes a crash due to double free)
+        releaseResourcesBackToPool();
+        return IGPUCommandBuffer::resetCommon();
+    }
```

Comment:
you're crashing because you shouldn't free resources of the pool if the pool has been reset

Reply:
Resolved.